### DISCUSSION TITLE
Working with Video Files

### DISCUSSION THREAD

**Discussion by JDebono2023** [awcodes/filament-curator#558] [Q&A]
I updated the accepted file types in the config, and was able to upload the required mp4. I normally process videos using FFMpeg (pbmedia/laravel-ffmpeg), and also collect the dimensions of the video (work specific requirement), as well as need to generate a url and thumbnail. Is there a way to add this sort of functionality in?

**Comment by awcodes** [comment]
Yes. But you would need a custom media upload action.

**Reply by JDebono2023** [reply]
I suspected so. Do you have any tips of how to best integrate that, with filament-curator, or should it be straight forward?

**Reply by awcodes** [reply]
So you're uploading a video with curator, then want to insert that into Tiptap?

**Reply by JDebono2023** [reply]
If that's the process that the plugin uses, then yes. Im not using TipTap in my project, specifically, and am not familiar with it. Im loading various media types to aws to store and add to playlists and then be used elsewhere in an external app.

**Comment by awcodes** [comment]
I totally misread this as a question for tiptap editor. sorry.

the way to handle this would be to extend the MediaObserver class and rebind it in a service provider to use your Observer instead of the one directly from the package.

**Reply by JDebono2023** [reply]
ok great. thanks for the info.
